### PR TITLE
fix(imports): #MC-138 fix hour change when importing ical

### PR DESCRIPTION
**Pull request by alicedraillard**
## Describe your changes
fix #88 
take into account events that last one whole day
fix based on jupyter dev : https://github.com/OPEN-ENT-NG/jupyter-connector/blob/87361918dd07e2630d3f0eb030e55772fdc032f8/src/main/java/fr/openent/jupyter/helper/DateHelper.java#L23

## Checklist tests
Import ics calendar from ticket

## Issue ticket number and link
#MC-138 https://entsupport.gdapublic.fr/browse/MC-138

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

**Review by aymericdevriesere — APPROVED** [review]

File: src/main/java/net/atos/entng/calendar/ical/ICalHandler.java [hunk 1]
```diff
@@ -331,4 +319,29 @@ private void addEvent(Calendar calendar, String startDate, String endDate, Strin
         }
     }
 
+    /* Return something that can handle errors ? */
+    private String formatIcsDate (java.util.Date date) throws UnhandledEventException {
```
Comment (BastienBenard) [inline src/main/java/net/atos/entng/calendar/ical/ICalHandler.java:323]: Tu peux passer cette methode en static

File: src/main/java/net/atos/entng/calendar/ical/ICalHandler.java [hunk 2]
```diff
@@ -331,4 +319,29 @@ private void addEvent(Calendar calendar, String startDate, String endDate, Strin
...
+        java.util.Date newStartDate;
+        try {
+            newStartDate = new SimpleDateFormat(DateUtils.ICAL_DATE_FORMAT).parse(date.toString());
+        } catch (ParseException e) {
+            newStartDate = formatIcsAlldayDate(date, e);
+        }
+        return new SimpleDateFormat(DateUtils.DATE_FORMAT_UTC).format(newStartDate);
+    }
+
+    private java.util.Date formatIcsAlldayDate(java.util.Date date, ParseException e1) throws UnhandledEventException {
```
Comment (BastienBenard) [inline src/main/java/net/atos/entng/calendar/ical/ICalHandler.java:333]: Tu peux passer cette methode en static

File: src/main/java/net/atos/entng/calendar/ical/ICalHandler.java [hunk 3]
```diff
@@ -331,4 +319,29 @@ private void addEvent(Calendar calendar, String startDate, String endDate, Strin
...
+        java.util.Date newStartDate;
+        try {
+            newStartDate = new SimpleDateFormat(DateUtils.ICAL_ALLDAY_FORMAT).parse(date.toString());
+        } catch (ParseException e2) {
+            String message1 = String.format("[Calendar@%s::formatIcsAlldayDate] Errors have occured" +
+                            " during ical import: %s and %s",
+                    this.getClass().getSimpleName(), e1.getMessage(), e2.getMessage());
```
Comment (BastienBenard) [inline src/main/java/net/atos/entng/calendar/ical/ICalHandler.java:340]: Si tu passe en static this.getClass().getSimpleName() -> ICalHandler.class.getSimpleName()

File: src/main/java/net/atos/entng/calendar/ical/ICalHandler.java [hunk 4]
```diff
@@ -331,4 +319,29 @@ private void addEvent(Calendar calendar, String startDate, String endDate, Strin
...
+            log.error(message1, e1.getMessage(), e2.getMessage());
```
Comment (qpic-cgi) [inline src/main/java/net/atos/entng/calendar/ical/ICalHandler.java:341]: concatène tes erreurs plutôt

**Review by BastienBenard — APPROVED** [review]

File: src/main/java/net/atos/entng/calendar/ical/ICalHandler.java [hunk 5]
```diff
@@ -335,4 +323,41 @@ private void addEvent(Calendar calendar, String startDate, String endDate, Strin
...
+     */
+    private static java.util.Date formatIcsAlldayDate(java.util.Date date, ParseException e1) throws UnhandledEventException {
+        java.util.Date newStartDate;
+        try {
+            newStartDate = new SimpleDateFormat(DateUtils.ICAL_ALLDAY_FORMAT).parse(date.toString());
+        } catch (ParseException e2) {
+            String finalError = String.format(e1.getMessage(), e2.getMessage());
+            String message1 = String.format("[Calendar@%s::formatIcsAlldayDate] Errors have occured during ical import: %s",
+                    ICalHandler.class.getSimpleName(), finalError);
+            log.error(message1, finalError);
```
Comment (qpic-cgi) [inline src/main/java/net/atos/entng/calendar/ical/ICalHandler.java:357]: Il sert a quoi finalError ici ? je pense que tu peux l'enlever étant donné qu'il est déjà présent dans ton message1.
Il serait d'ailleurs préférable de lui donner un autre nom que message1, comme par exemple messageError qui est plus déterminant.

File: src/main/java/net/atos/entng/calendar/ical/ICalHandler.java [hunk 6]
```diff
@@ -335,4 +323,40 @@ private void addEvent(Calendar calendar, String startDate, String endDate, Strin
...
+     * @return {@link java.util.Date} the date
+     * @throws UnhandledEventException
+     */
+    private static java.util.Date formatIcsAlldayDate(java.util.Date date, ParseException e1) throws UnhandledEventException {
+        java.util.Date newStartDate;
+        try {
+            newStartDate = new SimpleDateFormat(DateUtils.ICAL_ALLDAY_FORMAT).parse(date.toString());
+        } catch (ParseException e2) {
+            String errorMessage = String.format("[Calendar@%s::formatIcsAlldayDate] Errors have occured during ical import: %s",
+                    ICalHandler.class.getSimpleName(), String.format("%s %s", e1.getMessage(), e2.getMessage()));
```
Comment (qpic-cgi) [inline src/main/java/net/atos/entng/calendar/ical/ICalHandler.java:355]: tu n'es pas obligé de faire 2 string format, un seul suffit 😜

**Review by qpic-cgi — APPROVED** [review]

**Review by qpic-cgi — APPROVED** [review]

File: build.sh [hunk 7]
```diff
@@ -38,36 +38,36 @@ clean () {
 
 buildNode () {
   #jenkins
-  echo "[buildNode] Get branch name from jenkins env..."
```
Comment (alicedraillard) [inline build.sh:41]: commit will be deleted, used for ng2

File: src/main/java/net/atos/entng/calendar/ical/ICalHandler.java [hunk 8]
```diff
@@ -335,4 +323,40 @@ private void addEvent(Calendar calendar, String startDate, String endDate, Strin
...
+     */
+    private static java.util.Date formatIcsAlldayDate(java.util.Date date, ParseException e1) throws UnhandledEventException {
+        java.util.Date newStartDate;
+        try {
+            newStartDate = new SimpleDateFormat(DateUtils.ICAL_ALLDAY_FORMAT).parse(date.toString());
+        } catch (ParseException e2) {
+            String errorMessage = String.format("[Calendar@%s::formatIcsAlldayDate] Errors have occured during ical import: %s %s",
+                    ICalHandler.class.getSimpleName(), e1.getMessage(), e2.getMessage());
+            log.error(errorMessage);
+            throw new UnhandledEventException("calendar.ical.event.slot.problem");
```
Comment (alicedraillard) [inline src/main/java/net/atos/entng/calendar/ical/ICalHandler.java:357]: throw unhandledevent exception to replicate original use and get failted event imports
https://github.com/OPEN-ENT-NG/calendar/blob/27f93a92411548230574657d0e00b611e50a5089/src/main/java/net/atos/entng/calendar/ical/ICalHandler.java#L150

**Review by qpic-cgi — APPROVED** [review]

**Review by aymericdevriesere — APPROVED** [review]

**Pull request merged**